sway/commands/input/xkb_switch_layout.c

```diff
@@ -77,12 +91,23 @@ struct cmd_results *input_cmd_xkb_switch_layout(int argc, char **argv) {
 		if (dev->wlr_device->type != WLR_INPUT_DEVICE_KEYBOARD) {
 			continue;
 		}
+
+		struct xkb_switch_layout_action *action =
+			&actions[actions_len++];
+
+		action->keyboard = dev->wlr_device->keyboard;
 		if (relative) {
-			switch_layout_relative(dev->wlr_device->keyboard, relative);
+			action->layout = get_layout_relative(
+				dev->wlr_device->keyboard, relative);
 		} else {
-			switch_layout(dev->wlr_device->keyboard, layout);
+			action->layout = layout;
 		}
 	}
 
+	for (size_t i = 0; i < actions_len; i++) {
```

Comment:
Can we add a small comment somewhere explaining that we need a two-step process because switching a layout in one keyboard may result in a change on other keyboards as well because of keyboard groups?

Reply:
fixed